src/ansi-c/c_typecheck_code.cpp

```diff
@@ -805,16 +807,28 @@ void c_typecheck_baset::typecheck_dowhile(code_dowhilet &code)
   break_is_allowed=old_break_is_allowed;
   continue_is_allowed=old_continue_is_allowed;
 
-  typecheck_spec_expr(code, ID_C_spec_loop_invariant);
+  typecheck_loop_invariant(code);
+  typecheck_spec_decreases(code);
 }
 
-void c_typecheck_baset::typecheck_spec_expr(codet &code, const irep_idt &spec)
+void c_typecheck_baset::typecheck_loop_invariant(codet &code)
```

Comment:
[nit]

For consistency, can we name this `typecheck_spec_loop_invariant`?

So we would have `ID_C_spec_loop_invariant` checked by `typecheck_spec_loop_invariant` and `ID_C_spec_decreases` checked by `typecheck_spec_decreases`.

Reply:
Thanks for the suggestion!